Assets/MixedRealityToolkit/_Core/Devices/OpenVR/GenericOpenVRController.cs

```diff
@@ -199,337 +262,214 @@ protected void UpdatePointerData(MixedRealityInteractionMapping interactionMappi
+                case DeviceInputType.TriggerNearTouch:
+                case DeviceInputType.TriggerPress:
+                    // Get the current input state
+                    UpdateButtonFromAxis(interactionMapping);
+                    break;
+            }
+        }
 
-                    //Update the interaction data source
-                    interactionMapping.FloatData = triggerAxis;
+        /// <summary>
+        /// Update the buttons state.
+        /// </summary>
+        /// <param name="interactionMapping"></param>
+        protected void UpdateButtonData(MixedRealityInteractionMapping interactionMapping)
+        {
+            Debug.Assert(interactionMapping.InputType == DeviceInputType.ButtonPress);
 
-                    // If our value changed raise it.
-                    if (interactionMapping.Changed)
-                    {
-                        //Raise input system Event if it enabled
-                        InputSystem?.RaiseOnInputPressed(InputSource, ControllerHandedness, interactionMapping.MixedRealityInputAction, triggerAxis);
-                    }
+            // Get the current input state
+            UpdateButtonFromKey(interactionMapping);
+        }
+
+        /// <summary>
+        /// Update the Oculus Touch Thumb rest Data.
+        /// </summary>
+        /// <param name="interactionMapping"></param>
+        protected void UpdateOculusTouchThumbRestData(MixedRealityInteractionMapping interactionMapping)
+        {
+            switch (interactionMapping.InputType)
+            {
+                case DeviceInputType.ThumbTouch:
+                    // Get the current input state
+                    UpdateButtonFromKey(interactionMapping);
                     break;
-                default:
+                case DeviceInputType.ThumbNearTouch:
+                    // Get the current input state
+                    UpdateButtonFromAxis(interactionMapping);
                     break;
             }
         }
 
         /// <summary>
-        /// Update the buttons state.
+        /// Update the Valve Knuckles Finger Data.
         /// </summary>
-        /// <param name="interactionSourceState"></param>
         /// <param name="interactionMapping"></param>
-        protected virtual void UpdateButtonData(MixedRealityInteractionMapping interactionMapping)
+        protected void UpdateKnucklesFingerData(MixedRealityInteractionMapping interactionMapping)
         {
-            //Get the current Menu button state
-            var menuButton = ControllerHandedness == Handedness.Left ? Input.GetKey(KeyCode.JoystickButton2) : Input.GetKey(KeyCode.JoystickButton0);
+            UpdateSingleAxisFromAxis(interactionMapping);
+        }
+
+        #endregion Update data functions
+
+        #region Event Functions
+
+        protected void UpdateSingleAxisFromAxis(MixedRealityInteractionMapping interactionMapping)
```

Comment:
What's the difference from this and UpdateButtonFromAxis?

I do like that it's a bit cleaner.